pkg/nfd-master/node-updater-pool.go

```diff
@@ -0,0 +1,74 @@
+package nfdmaster
+
+import (
+	"sync"
+
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+type nodeUpdaterPool struct {
+	queue     workqueue.RateLimitingInterface
+	wg        sync.WaitGroup
+	nfdMaster nfdMaster
+}
+
+var lock = &sync.Mutex{}
+var instance *nodeUpdaterPool
+
+func getNodeUpdaterPool(nfdMaster nfdMaster) *nodeUpdaterPool {
+	if instance == nil {
+		lock.Lock()
+		defer lock.Unlock()
+		if instance == nil {
+			instance = &nodeUpdaterPool{
+				nfdMaster: nfdMaster,
+				wg:        sync.WaitGroup{},
+			}
+		}
+	}
+	return instance
+}
+
+func (u *nodeUpdaterPool) processNodeLabelRequest(queue workqueue.RateLimitingInterface) bool {
+	nodeName, quit := queue.Get()
+	if quit {
+		return false
+	}
+
+	defer queue.Done(nodeName)
+
+	if err := u.nfdMaster.nfdAPIUpdateOneNode(nodeName.(string)); err != nil {
```

Comment:
and if this is the only place where `nfdAPIUpdateOneNode` is called you can just remove this func from `nfdMaster` and make it `nodeUpdaterPoll` func, this way you don't need to pass `nfdMaster` to `nodeUpdaterPool` but prolly just Kubernetes client.

Reply:
Ok, we should create issue to track this work.